File: Medium/Implement Atoi/implement-atoi.java
//{ Driver Code Starts
//Initial template for JAVA

import java.util.Scanner;

class aToi
{
	public static void main(String args[])
	{
		Scanner sc = new Scanner(System.in);
		int t = sc.nextInt();
		sc.nextLine();
		while(t>0)
		{
			String str = sc.nextLine();
			
			Solution obj = new Solution();
			int num = obj.atoi(str);
			
			System.out.println(num);
		t--;
		}
	}
}
// } Driver Code Ends


//User function template for JAVA

/*You are required to complete this method */
class Solution
{
    int atoi(String s) {
	// Your code here
	    s = s.trim();
	    if(s.isEmpty()){
	        return -1;
	    }
	    int result = 0;
	    boolean flag = false;
	    if(s.charAt(0) == '-'){
	        flag = true; 
	    }
	    int neg = 0;
	    if(flag){
	        neg += 1;
	    }
	    for(int i=neg;i<s.length();i++){
	        if(s.charAt(i) >= '0' && s.charAt(i)<= '9'){
	            int digit = s.charAt(i) - '0';
	            result = result*10 + digit;
	        }
	        else{
	            return -1;
	        }
	    }
	    if(flag){
	        return result*-1;
	    }
	    return result;
    }
}

File: Difficulty: Medium/Implement Atoi/implement-atoi.cpp
//{ Driver Code Starts
#include <bits/stdc++.h>
using namespace std;


// } Driver Code Ends
class Solution {
  public:
    int myAtoi(char *s) {
        // Your code here
        
        // s = s.trim();
        
        int n = strlen(s);
        
        if(n == 0){
            return 0;
        }
        
        int neg = 0;
        
        if(s[0] == '-'){
            neg++;
        }
        
        long long ans = 0;
        
        for(int i=neg;i<n;i++){
            if(s[i] == ' '){
                continue;
            }
            if(s[i] >= '0' && s[i] <= '9'){
                int digit = s[i] - '0';
                ans = ans*10 + digit;
            }
            else{
                break;
            }
            
        }
        
        if(s[0] == '-'){
            ans = -1*ans;
        }
        
        if(ans <= INT_MIN){
            return INT_MIN;
        }
        if(ans >= INT_MAX){
            return INT_MAX;
        }
        
        return ans;
        
    }
};

//{ Driver Code Starts.
int main() {
    int t;
    cin >> t;
    while (t--) {
        char s[20];
        cin >> s;
        Solution ob;
        int ans = ob.myAtoi(s);
        cout << ans << endl;
        cout << "~" << endl;
    }
}
// } Driver Code Ends